go.mod

```diff
@@ -1,29 +1,133 @@
 module github.com/crossplane/crossplane
 
-go 1.16
+go 1.17
 
 require (
-	github.com/Azure/go-autorest/autorest/validation v0.2.0 // indirect
 	github.com/Masterminds/semver v1.5.0
 	github.com/alecthomas/kong v0.2.17
-	github.com/aws/aws-sdk-go v1.31.6 // indirect
-	github.com/crossplane/crossplane-runtime v0.15.1-0.20211029211307-c72bcdd922eb
-	github.com/docker/cli v0.0.0-20200915230204-cd8016b6bcc5 // indirect
-	github.com/docker/docker v17.12.0-ce-rc1.0.20200926000217-2617742802f6+incompatible // indirect
+	github.com/crossplane/crossplane-runtime v0.15.1-0.20220106140106-428b7c390375
 	github.com/google/go-cmp v0.5.6
-	github.com/google/go-containerregistry v0.4.1
-	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20210330174036-3259211c1f24
+	// TODO(hasheddan): we prefer to consume release versions of
+	// go-containerregistry. An incremental version is currently being used to
+	// consume the new k8schain implementation, which fixes an issue with
+	// reconcile timeouts when pulling packages in some environments.
+	github.com/google/go-containerregistry v0.8.1-0.20220120151853-ac864e57b117
+	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20220120151853-ac864e57b117
 	github.com/imdario/mergo v0.3.12
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/afero v1.6.0
-	golang.org/x/tools v0.1.5
-	k8s.io/api v0.21.3
-	k8s.io/apiextensions-apiserver v0.21.3
-	k8s.io/apimachinery v0.21.3
-	k8s.io/client-go v0.21.3
-	k8s.io/code-generator v0.21.3
-	k8s.io/utils v0.0.0-20210722164352-7f3ee0f31471
-	sigs.k8s.io/controller-runtime v0.9.6
-	sigs.k8s.io/controller-tools v0.6.2
-	sigs.k8s.io/yaml v1.2.0
+	github.com/sirupsen/logrus v1.8.1
+	github.com/spf13/afero v1.8.0
+	golang.org/x/tools v0.1.8
+	k8s.io/api v0.23.0
+	k8s.io/apiextensions-apiserver v0.23.0
+	k8s.io/apimachinery v0.23.0
+	k8s.io/client-go v0.23.0
+	k8s.io/code-generator v0.23.0
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
+	sigs.k8s.io/controller-runtime v0.11.0
```

Comment:
Hrm, do we really want to go from 0.9.x -> 0.11.x of this fairly impactful dependency in a patch release? 🤔 Hypothetically there are similar concerns for crossplane-runtime, but I'm fairly sure not much has changed in that bump.

Reply:
@negz fair point -- I included it as it was grouped with the `1.17` change, but I believe we could drop https://github.com/crossplane/crossplane/pull/2852/commits/5bf6faad9da505bbcd2e9ed02c4b3ff0e1b241b5 and it wouldn't complicate much.

Reply:
@negz it does introduce some conflicts -- with them resolved it looks like this: https://github.com/hasheddan/crossplane/tree/rel-1.6-k8schain-rebase

Definitely agree that bumping two versions of `controller-runtime` feels out of place for a patch. I did look through the releases and the changes mentioned do not look overly concerning to me (https://github.com/kubernetes-sigs/controller-runtime/releases -- the fairly significant refactor of `Manager` does give me pause). That being said, I think I might opt to drop the backport strategy altogether so we don't have to pick up the `1.17` changes and instead just make the same changes from https://github.com/crossplane/crossplane/pull/2841 on `release-1.6`. What do you think?

Reply:
@negz this turned out to be more of a challenge than expected as the `go-containerregistry` updates bump [k8s deps](https://github.com/google/go-containerregistry/blob/ac864e57b117d74a16269906dcbbd8db366eaadd/pkg/authn/kubernetes/go.mod#L9), which in turn break logging in our pinned version of `controller-runtime` (due to `client-go` changes). We could try to use a `replace` and find a common version that works, but that honestly feels more brittle to me than just moving ahead with the bumps.

Reply:
@hasheddan If you feel comfortable I could be convinced to learn to stop worrying and love the bump. ;) Maybe we can kick the tyres a bit before we release it, just to make sure there's nothing weird with the manager refactor? The composition controllers in particular use some fairly obscure features of the `Manager` code.

Reply:
@negz I am feeling pretty good about it (well, relative to the alternatives). The `controller-runtime` release has had about a month of bake time and I have at least given a cursory review of most of the major changes. Definitely agree that we should do some further testing before release. I specifically want to test some of the heavy load scenarios and will likely enlist some community members to try to stress it in their org's dev environments. I think a release early next week would be plenty of time for me to feel confident in the stability of these bumps.